File: Módulo Python/2 - Estrutura Sequencial/Exercicio.27.py
#27. A Empresa ABC produz 3 tipos de peças mecânicas: parafusos, porcas e arruelas. Sabe-se que é dado desconto de:
# 10% por porca vendida, 20% por Arruelas vendidas, 30% por parafusos vendidos. Dados de Entrada: nome do cliente; 
# preço unitário da porca, arruela, parafuso; quantidade de porcas, arruelas e parafusos solicitados pelo cliente. 
# Dados de Saída: nome do cliente; número de porcas, arruelas e parafusos comprados; total pago pelas porcas, arruelas 
# e parafusos; total de desconto dado ao cliente; total a ser pago pelo cliente.

nome = input("Digite o nome do cliente: ")
precoPorca = float(input("Digite o preço unitário da porca: "))
precoArruela = float(input("Digite o preço unitário da arruela: "))
precoParafuso = float(input("Digite o preço unitário do parafuso: "))
qntdPorcas = int(input("Digite a quantidade de porcas: "))
qntdArruelas = int(input("Digite a quantidade de arruelas: "))
qntdParafusos = int(input("Digite a quantidade de parafusos: "))
totalPorcas = qntdPorcas * precoPorca
totalArruelas = qntdArruelas * precoArruela
totalParafusos = qntdParafusos * precoParafuso
descontoPorcas = totalPorcas * 0.1
descontoArruelas = totalArruelas * 0.2
descontoParafusos = totalParafusos * 0.3
totalDesconto = descontoPorcas + descontoArruelas + descontoParafusos
total = totalPorcas - descontoPorcas + totalArruelas - descontoArruelas + totalParafusos - descontoParafusos

print("Nome do cliente:", nome)
print("Quantidade de porcas compradas:", qntdPorcas)
print("Quantidade de arruelas compradas:", qntdArruelas)
print("Quantidade de parafusos comprados:", qntdParafusos)
print("Total pago pelas porcas: R$", totalPorcas)
print("Total pago pelas arruelas: R$", totalArruelas)
print("Total pago pelos parafusos: R$", totalParafusos)
print("Total de desconto dado ao cliente: R$", totalDesconto)
print("Total a ser pago pelo cliente: R$", total)

File: Livro - Módulo Java/2 - Estrutura Sequencial/Exercicio.27.java
// 27 - A Empresa ABC produz 3 tipos de peças mecânicas: parafusos, porcas e arruelas. 
// Sabe-se que é dado desconto de 10% sobre as porcas vendidas, de 20% sobre as arruelas
// // vendidas e de 30% sobre os parafusos vendidos. (L1 – ex.24) 
// O algoritmo deve receber os seguintes dados: 
// - nome do cliente; 
// - preço unitário da porca, arruela e parafuso; 
// - quantidade de porcas, arruelas e parafusos solicitados pelo cliente. 
// Por fim, deve-se imprimir os seguintes dados: 
// - nome do cliente; 
// - número de porcas, arruelas e parafusos comprados; 
// - total pago pelas porcas, arruelas e parafusos; 
// - total de desconto dado ao cliente; 
// - total a ser pago pelo cliente. 

import java.util.Scanner;

public class Ex24Scanner
{

    public static void main(String[] args)
    {
        String nome;
        float unitPorca;
        float unitArruela;
        float unitParafuso;
        int qtdPorca;
        int qtdArruela;
        int qtdParafuso;
        float precoPorca;
        float precoArruela;
        float precoParafuso;
        float desconto;
        float precoTotal;

        Scanner input = new Scanner(System.in);

        System.out.println("Qual o nome do cliente?");
        nome = input.nextLine();
        System.out.println("Qual o preço da porca?");
        unitPorca = input.nextFloat();
        System.out.println("Qual o preço da arruela?");
        unitArruela = input.nextFloat();
        System.out.println("Qual o preço da parafuso?");
        unitParafuso = input.nextFloat();
        System.out.println("Quantas porcas foram compradas?");
        qtdPorca = input.nextInt();
        System.out.println("Quantas arruelas foram compradas?");
        qtdArruela = input.nextInt();
        System.out.println("Quantos parafusos foram compradas?");
        qtdParafuso = input.nextInt();

        precoPorca = qtdPorca * unitPorca;
        precoArruela = qtdArruela * unitArruela;
        precoParafuso = qtdParafuso * unitParafuso;

        precoTotal = precoPorca + precoArruela + precoParafuso;

        desconto = precoTotal - ((precoPorca * 0.9f) + (precoArruela * 0.8f) + (precoParafuso * 0.7f));

        System.out.printf("O cliente %s comprou %d porcas por R$%.2f, %d arruelas por R$%.2f e %d parafusos por R$%.2f, com desconto de R$%.2f, totalizando R$%.2f.",
                nome, qtdPorca, precoPorca, qtdArruela, precoArruela, qtdParafuso, precoParafuso, desconto, precoTotal);
    }
}